codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGeneration.kt

```diff
@@ -0,0 +1,281 @@
+            }.joinToString(separator = "\n")
+            val registerOperationsBuilder = operationNames.map { operationName ->
+                """
+                $operationName: match self.$operationName {
+                    Some(v) => v,
+                    None => return Err(${operationRegistryBuilderName}Error::UninitializedField(${operationName.dq()})),
+                }
+                """
+            }.joinToString(separator = "\n,")
+            rustTemplate(
+                """
+                $registerOperations
+                pub fn build(self) -> Result<$operationRegistryNameWithArguments, ${operationRegistryBuilderName}Error> {
+                    Ok($operationRegistryName { $registerOperationsBuilder, _phantom: #{Phantom} })
+                }
+                """,
+                *codegenScope
+            )
+        }
+    }
+
+    /*
+     * Renders the conversion between the OperationRegistry and the Router via the into() method.
+     */
+    private fun renderRouterImplFromOperationRegistryBuilder(writer: RustWriter) {
+        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation
+        val operationsTraitBounds = operations
+            .mapIndexed { i, operation ->
+                val operationName = symbolProvider.toSymbol(operation).name
+                """Op$i: crate::operation_handler::Handler<B, In$i, crate::input::${operationName}Input>,
+            In$i: 'static"""
+            }.joinToString(separator = ",\n")
+        Attribute.Custom("allow(clippy::all)").render(writer)
+        writer.rustBlockTemplate(
+            """
+            impl<$genericArguments> From<$operationRegistryNameWithArguments> for #{Router}<B>
+            where
+                B: Send + 'static,
+                $operationsTraitBounds
+            """.trimIndent(),
+            *codegenScope
+        ) {
+            rustBlock("fn from(registry: $operationRegistryNameWithArguments) -> Self") {
+                val requestSpecsVarNames = operationNames.map { "${it}_request_spec" }
+                val routes = requestSpecsVarNames.zip(operationNames) { requestSpecVarName, operationName ->
+                    ".route($requestSpecVarName, crate::operation_handler::operation(registry.$operationName))"
+                }.joinToString(separator = "\n")
+
+                val requestSpecs = requestSpecsVarNames.zip(operations) { requestSpecVarName, operation ->
+                    "let $requestSpecVarName = ${operation.requestSpec()};"
+                }.joinToString(separator = "\n")
+                rustTemplate(
+                    """
+                    $requestSpecs
+                    #{Router}::new()
+                        $routes
+                    """.trimIndent(),
+                    *codegenScope
+                )
+            }
+        }
+    }
+
+    /*
+     * Renders the PhantomData generic members.
+     */
+    private fun phantomMembers(): String {
```

Comment:
Let's move this function closer to where it's used.

Reply:
Why? I jump to the definition using my editor. This change doesn't add any value.